File: D2/D2.java
import java.util.*;

public class D2 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int T = scanner.nextInt(); // Number of test cases
        for (int t = 1; t <= T; t++) {
            int N = scanner.nextInt(); // Number of stones
            int G = scanner.nextInt(); // Goal
            int[] energies = new int[N];
            for (int i = 0; i < N; i++) {
                energies[i] = scanner.nextInt();
            }

            // List to store the final positions of the stones
            int[] finalPositions = new int[N];
            Set<Integer> occupied = new HashSet<>(); // Use a set to track occupied positions

            // Simulating the stone throws
            for (int i = 0; i < N; i++) {
                int energy = energies[i];
                int position = 0; // Starting at position 0

                // Move the stone until its energy is exhausted or it collides with another
                while (energy > 0) {
                    if (occupied.contains(position + 1)) {
                        // If the next position is occupied, break to simulate collision
                        break;
                    } else {
                        // Move the stone one unit to the right
                        position++;
                        energy--;
                    }
                }

                // The stone is now stationary at position
                finalPositions[i] = position;
                occupied.add(position); // Mark this position as occupied
            }

            // Find the stone closest to the goal G
            Integer closestStone = null;
            int minDistance = Integer.MAX_VALUE;

            for (int i = 0; i < N; i++) {
                int distance = Math.abs(finalPositions[i] - G);
                if (distance < minDistance || (distance == minDistance && (closestStone == null || i < closestStone))) {
                    minDistance = distance;
                    closestStone = i; // Store index of closest stone
                }
            }

            System.out.printf("Case #%d: %d %d%n", t, closestStone + 1, minDistance); // Convert to 1-based index
        }
        scanner.close();
    }
}

File: D2/D2.py
def solve():
    T = int(input())  # Number of test cases
    for t in range(1, T + 1):
        N, G = map(int, input().split())  # N is number of stones, G is the goal
        energies = [int(input()) for _ in range(N)]

        # List to store the final positions of the stones
        final_positions = [0] * N
        occupied = set()  # Use a set to track occupied positions

        # Simulating the stone throws
        for i in range(N):
            energy = energies[i]
            position = 0  # Starting at position 0

            # Move the stone until its energy is exhausted or it collides with another
            while energy > 0:
                if position + 1 in occupied:
                    # If the next position is occupied, break to simulate collision
                    break
                else:
                    # Move the stone one unit to the right
                    position += 1
                    energy -= 1
            
            # The stone is now stationary at position
            final_positions[i] = position
            occupied.add(position)  # Mark this position as occupied
        
        # Find the stone closest to the goal G
        closest_stone = None
        min_distance = float('inf')

        for i in range(N):
            distance = abs(final_positions[i] - G)
            if distance < min_distance or (distance == min_distance and (closest_stone is None or i < closest_stone)):
                min_distance = distance
                closest_stone = i  # Store index of closest stone
        
        print(f"Case #{t}: {closest_stone + 1} {min_distance}")  # Convert to 1-based index

